### ISSUE TITLE
Replace `Guard` with `TypeOf`

### ISSUE THREAD

**Issue by sellout** [precog/quasar#1546]
`Guard(expr, type, consequent, alternative)` can be broken down into `Cond(TypeOf(expr) = type, consequent, alternative)`. This isn’t quite as rich as `Guard` (but much easier for connectors to implement.

**Comment by zeeiyerWork** [comment]
Check to see if this is a duplicate of another issue.

**Comment by alissapajer** [comment]
#1551 gets us the `LogicalPlan` change from `LP.Typecheck` to `LP.Match` (or something that avoids the duplicate conditional logic) and then #1552 compiles that `LP.Match` (or whatever we end up with) to `QScript`. I don't expect we'll merge #1551 without #1552.

This ticket is about how we represent typechecking in `QScript`, unrelated to `LogicalPlan`. It will probably be easiest to make this change at the same as #1552 but technically we don't have to.

**Comment by alissapajer** [comment]
Should probably be postponed until after #2567, unless there is a connector-driven need to get this done sooner.
